pages/[[...slug]]/index.tsx

```diff
@@ -0,0 +1,35 @@
+import { getAllPathnames, getPageContentByPathname } from '../../utilities/api'
+import type { GetStaticPaths, GetStaticProps } from 'next'
+import PageContent from '../../components/Main/PageContent'
+import type { PageContentProps } from '../../components/Main/PageContent'
+import React from 'react'
+import type { Pathname } from '../../utilities/api'
+
+type SitePageProps = {
+  pageContent: PageContentProps
+}
+
+export default function SitePage({ pageContent }: SitePageProps): JSX.Element {
+  return <PageContent content={pageContent.content} title={pageContent.title} />
+}
+
+export const getStaticProps: GetStaticProps = async ({ params }) => {
+  const pathname: string = (params?.slug as string[])?.join('/') ?? 'index'
+  const pageContent = await getPageContentByPathname(pathname)
+
+  return { props: { pageContent } }
+}
+
+export const getStaticPaths: GetStaticPaths = async () => {
+  type ParamWithSlug = { params: { slug: string[] } }
```

Comment:
Never seen this before. Are we creating a type in the body of the function here?

Could this be at the top of the file with the other types?

Reply:
This type was `type Path = { params: { slug: string[] } }` before. Its a Next thing just for the `getStaticPaths()`, so it was put inside of the function. 
It was renamed because I thought `Path` should mean a string, not an object. I can't change the `paths` in the returned object, so I renamed the type. 
I come up with another way to get rid of this type. May be clearer:
```ts
export const getStaticPaths: GetStaticPaths = async () => {
  // type ParamWithSlug = { params: { slug: string[] } }

  const allPathnames = (await getAllPathnames()) as Pathname[]
  const allSlugs: { slug: string[] }[] = allPathnames.map(({ pathname }) => ({
    slug: pathname !== 'index' ? pathname.split('/') : [],
  }))

  // const paths: ParamWithSlug[] = allPathnames.map(({ pathname }) => ({
  //   params: { slug: pathname !== 'index' ? pathname.split('/') : [] },
  // }))

  return {
    fallback: false,
    paths: allSlugs.map((slug) => ({ params: slug })),
  }
}
```
How do you feel about it?

Reply:
This is not a valid discussion now. Thanks for the advice.